### DISCUSSION TITLE
Unable to create tree-view for a collection

### DISCUSSION THREAD

**Discussion by ntsh999** [directus/directus#13376] [Archive]
I want to create a tree-view (using the tree view interface) to visualize the parent child relationship existing in a collection. A Parent can have one or more children. But children will not have more sub-children. Is there any documentation available that helps in achieving the required objective? Can someone help and guide in achieving the same?

### To Reproduce

Create a collection named as "Categories"
1. id - PK, manually entered string
2. name - category name
3. parent - M20 relationship with name column. Parent will be null/empty if the category is at the topmost level. Otherwise if the category is not at the topmost level, it will have the id of the category.

### What version of Directus are you using?

9.8.0

**Comment by azrikahar** [comment]
Hi! Would you mind updating to the latest version `9.10.0`? There has been a major relational interface rework back in `9.9.0` which did include the tree view interface so things can be pretty different 👍

>  But children will not have more sub-children.

You may be able to achieve this with field conditions and disable/hide it when parent is null 🤔

**Reply by ntsh999** [reply]
@azrikahar I am trying to create a tree view using directus v9.10.0 but still unable to achieved what I need. My "categories" collection has 3 columns ( see "To Reproduce" at the top). Which column should I create using the tree view interface? Which column should be related to another column in the categories collection and with what type of relationship?

**Reply by azrikahar** [reply]
You can type in, instead of selecting, a preferred foreign key name for it to automatically create the column for you! Here's a quick gif:

![chrome_SKJy2HyYt0](https://user-images.githubusercontent.com/42867097/169775786-e8b3a010-6d51-4ac3-8fd7-260b0afde2bb.gif)

**Comment by leosin** [comment]
How do you tell if the parent is null or empty? @azrikahar

**Reply by azrikahar** [reply]
Using [the gif above](https://github.com/directus/directus/discussions/13376#discussioncomment-2802662) as an example, the `parent_test` field should be null if the item is not a child, unless I'm misunderstanding something here 🤔
